spec/services/pulfa_exporter_spec.rb

```diff
@@ -0,0 +1,93 @@
+  let(:temp_ead) { Rails.root.join(eads_dir, "C0652.EAD.xml") }
+  let(:collection_code) { "C0652" }
+  let(:component_code) { "c0377" }
+  let(:component_code2) { "c0383" }
+  let(:component_id) { "#{collection_code}_#{component_code}" }
+  let(:component_id2) { "#{collection_code}_#{component_code2}" }
+  let(:change_set_persister) { ChangeSetPersister.new(metadata_adapter: Valkyrie.config.metadata_adapter, storage_adapter: Valkyrie.config.storage_adapter) }
+  let(:collection) { FactoryBot.create_for_repository(:collection, source_metadata_identifier: collection_code) }
+  let(:resource) do
+    r = FactoryBot.build(:complete_scanned_resource, title: [])
+    change_set = ScannedResourceChangeSet.new(r)
+    change_set.validate(source_metadata_identifier: component_id, state: ["complete"], member_of_collection_ids: [collection.id])
+    change_set_persister.save(change_set: change_set)
+  end
+  let(:resource2) do
+    r = FactoryBot.build(:complete_scanned_resource, title: [])
+    change_set = ScannedResourceChangeSet.new(r)
+    change_set.validate(source_metadata_identifier: component_id2, state: ["complete"], member_of_collection_ids: [collection.id])
+    change_set.validate(member_ids: [vol1.id, vol2.id])
+    change_set_persister.save(change_set: change_set)
+  end
+  let(:vol1) { FactoryBot.create_for_repository(:scanned_resource, title: "Volume 1") }
+  let(:vol2) { FactoryBot.create_for_repository(:scanned_resource, title: "Volume 2") }
+
+  let(:ns) { { xlink: "http://www.w3.org/1999/xlink", ead: "urn:isbn:1-931666-22-9" } }
+  let(:xpath) { "//ead:dao[@xlink:role='https://iiif.io/api/presentation/2.1/']/@xlink:href" }
+  let(:xpath2) { "//ead:dao/@xlink:href" }
+
+  before do
+    FileUtils.mkdir_p(eads_dir) unless File.directory?(eads_dir)
+    FileUtils.cp(fixture_ead, eads_dir)
+    stub_pulfa(pulfa_id: component_id)
+    stub_pulfa(pulfa_id: component_id2)
+    stub_ezid(shoulder: "99999/fk4", blade: "8675309")
+    resource
+    resource2
+  end
+
+  after do
+    FileUtils.rm_rf(svn_dir)
+  end
+
+  describe "#export" do
+    let(:exporter) { described_class.new(since_date: since_date, logger: logger, svn_client: svn_client) }
+    let(:since_date) { (Time.zone.today - 2).strftime("%Y-%m-%d") }
+
+    it "adds a DAO link to the new resource" do
+      before = Nokogiri::XML(File.open(temp_ead))
+      expect(before.at_xpath(xpath, ns).to_s).to eq ""
+
+      expect { exporter.export }.not_to raise_error
+
+      after = Nokogiri::XML(File.open(temp_ead))
+      expect(after.at_xpath(xpath, ns).to_s).to eq "http://www.example.com/concern/scanned_resources/#{resource.id}/manifest"
+    end
+
+    describe "when there is an error sending email" do
+      let(:mailer) { instance_double(PulfaMailer) }
+
+      before do
+        allow(logger).to receive(:warn)
+        allow(PulfaMailer).to receive(:with).and_raise(StandardError, "No route to host")
+      end
+
+      it "catches and logs the error" do
```

Comment:
Should we have this report to honeybadger?

Reply:
see, e.g. https://github.com/pulibrary/figgy/blob/master/spec/jobs/update_fixity_job_spec.rb#L21-L27